### DISCUSSION TITLE
Apply Mask to Animation Layer

### DISCUSSION THREAD

**Discussion by diegostoliar** [keiwando/toonsquid#335] [Ideas]
We can toggle a layer to be masked by another layer, which is amazing, but if would be cool if I could apply the mask, effectively erasing what is already invisible, making my masked animation layer back into a normal animation layer, which allows me to use THAT layer as a mask for something else... of course, if there is tweening the layer would rasterized.... (OR maybe when we rasterize any layer it applies the mask automatically, erasing pixels out of it).

Here is use case:
I made a shadow layer, masked it no my character fill layer, but now I want that hard shadow to be masked by a gradient layer to create a shadow ramp effect.

Hope it makes sense.
Diego

**Comment by keiwando** [comment]
I see what you mean, thanks for the suggestion! Do you know of any other software that allows you to do this?

**Comment by diegostoliar** [comment]
I believe all raster based image editors do it… in procreate for example
you can merge the mask of a layer on itself, which applies the mask. I
guess the opportunity you have on toonsquid  is to do it fast for all
frames, which the other editors don’t do because of the way they layer
things.

On Wednesday, June 21, 2023, Keiwan Donyagard ***@***.***>
wrote:

> I see what you mean, thanks for the suggestion! Do you know of any other
> software that allows you to do this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/keiwando/toonsquid/discussions/335#discussioncomment-6245717>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNFHN5KKR6MS7QBPZCJXATXMNJUTANCNFSM6AAAAAAZCPA4OU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Cheers,
Diego Stoliar
www.stoliar.art

**Comment by keiwando** [comment]
I see. I initially thought you meant the ability to have a layer structure "A masked to B masked to C" where A is only masked to the visible portion of B instead of being masked to C which is currently the case if you have multiple masked layers in a row.
Reading the post and your reply again, I think I understood now that you mean having the option to keep the masked appearance of an animation layer while rasterizing it, so the mask determines what remains visible on the masked layer without having to merge both layers together. Is that right?

**Reply by diegostoliar** [reply]
'I think I understood now that you mean having the option to keep the masked appearance of an animation layer while rasterizing it, so the mask determines what remains visible on the masked layer without having to merge both layers together. Is that right?'

Yes, the new raster layer is the boolean result of the artwork minus the mask. Right now when I raster the layer I just get a copy of my original unmasked art... I guess what we are actually talking about is "raster" a layer should truly mean: apply anything affecting that layer (mask, blur, color, whatever) and give me a new bitmap sequence with the baked result.

Cool thing about this is: we can do very complex things, even if the app gets slow to play something eventually, and than bake things to keep interating with the result. Like the example before: mask a shadow shape with a character, bake it, and after mask a ramp gradient on top, bake it again and keep going.
